src/pynwb/icephys.py

```diff
@@ -319,15 +338,513 @@ def __get_row_ids(self, sweep_number):
+        if response.neurodata_type == 'IZeroClampSeries':
+            if stimulus is not None:
+                raise ValueError("stimulus should usually be None for IZeroClampSeries response")
+        if isinstance(response, PatchClampSeries) and isinstance(stimulus, PatchClampSeries):
+            # # We could also check sweep_number, but since it is mostly relevant to the deprecated SweepTable
+            # # we don't really need to enforce it here
+            # if response.sweep_number != stimulus.sweep_number:
+            #     warnings.warn("sweep_number are usually expected to be the same for PatchClampSeries type "
+            #                   "stimulus and response pairs in an intracellular recording.")
+            if response.electrode != stimulus.electrode:
+                raise ValueError("electrodes are usually expected to be the same for PatchClampSeries type "
+                                 "stimulus and response pairs in an intracellular recording.")
+
+        # Compile the electrodes table data
+        electrodes = copy(popargs('electrode_metadata', kwargs))
+        if electrodes is None:
+            electrodes = {}
+        electrodes['electrode'] = electrode
+
+        # Compile the stimuli table data
+        stimuli = copy(popargs('stimulus_metadata', kwargs))
+        if stimuli is None:
+            stimuli = {}
+        stimuli['stimulus'] = (stimulus_start_index, stimulus_index_count, stimulus)
+
+        # Compile the reponses table data
+        responses = copy(popargs('response_metadata', kwargs))
+        if responses is None:
+            responses = {}
+        responses['response'] = (response_start_index, response_index_count, response)
+
+        _ = super().add_row(enforce_unique_id=True,
+                            electrodes=electrodes,
+                            responses=responses,
+                            stimuli=stimuli,
+                            **kwargs)
+        return len(self) - 1
+
+    @staticmethod
+    def __compute_index(start_index, index_count, time_series, name):
+        """
+        Internal helper function to compute the start_index and index_count
+        to use for the stimulus and response column
+
+        :param start_index: The start_index provided by the user
+        :param index_count: The index count provided by the user
+        :param time_series: The timeseries object to reference. May be None.
+        :param name: Name of the table. Used only to enhance error reporting
+
+        :raises IndexError: If index_count cannot be determined or start_index+index_count
+            are outside of the range of the timeseries.
+
+        :returns: A tuple of integers with the start_index and index_count to use
+        """
+        if time_series is None:
+            return -1, -1
+        start_index = -1 if start_index is None else start_index
+        start_index = start_index if start_index >= 0 else 0
```

Comment:
```suggestion
        if start_index is None or start_index < 0:
            start_index = 0
```

Reply:
I updated this function to clarify the logic and add comments.